### DISCUSSION TITLE
tutorials/shaders/custom_postprocessing

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#260] [User-contributed notes]
# tutorials/shaders/custom_postprocessing

Introduction: Godot provides many post-processing effects out of the box, including Bloom, DOF, and SSAO, which are described in Environment and post-processing. However, advanced use cases may req...

https://docs.godotengine.org/en/stable/tutorials/shaders/custom_postprocessing.html

**Comment by SephReed** [comment]
Here's a discussion about various post-processing methods and pros and cons and such:
https://github.com/godotengine/godot/issues/99491

**Comment by ShinyAdmiral** [comment]
For anyone wondering struggling to find how to set to "Full Rect" in the ColorRect's Layout, its the "Anchors Preset" option.
(4.3)

**Reply by Calinou** [reply]
Pull request opened:

- https://github.com/godotengine/godot-docs/pull/10714
